sdk/PrebidMobile/PBBidManager.m

```diff
@@ -236,27 +232,26 @@ - (void)checkForBidsExpired {
     if (_adUnits != nil && _adUnits.count > 0) {
         NSTimeInterval currentTime = [[NSDate date] timeIntervalSince1970];
         for (PBAdUnit *adUnit in _adUnits) {
-            if ([adUnit shouldExpireAllBids:currentTime]) {
+            NSMutableArray *bids = [_bidsMap objectForKey:adUnit.identifier];
+            if (bids && [bids count] > 0 && [adUnit shouldExpireAllBids:currentTime]) {
                 [self startNewAuction:adUnit];
             }
         }
     }
 }
 
-- (PBBidResponse *)winningBidForAdUnit:(PBAdUnit *)adUnit {
+- (nullable NSArray<PBBidResponse *> *)getBids:(PBAdUnit *)adUnit {
     NSMutableArray *bids = [_bidsMap objectForKey:adUnit.identifier];
-    // can just take the first bid in the array because bids were already sorted server side
-    PBBidResponse *topBidResponse = bids[0];
-    if (topBidResponse && topBidResponse.isExpired == NO) {
-        [self startNewAuction:adUnit];
-    } else {
-        PBLogDebug(@"No bid responses for ad unit %@", adUnit.identifier);
+    if (bids && [bids count] > 0) {
+        return bids;
     }
-    return topBidResponse;
+    return nil;
 }
 
 - (BOOL)isBidReady:(NSString *)identifier {
-    if ([[_bidsMap allKeys] containsObject:identifier] && [_bidsMap objectForKey:identifier] != nil) {
+    if ([[_bidsMap allKeys] containsObject:identifier] &&
+        [_bidsMap objectForKey:identifier] != nil &&
+        [[_bidsMap objectForKey:identifier] count] > 0) {
```

Comment:
This is getting one object from _bidsMap, why checking the count here?

Reply:
It's just an extra check to make sure the _bidsMap isn't empty. The same check is done on Android in this function, I wanted to make them consistent.